src/Castle.Core/DynamicProxy/Generators/Emitters/MethodEmitter.cs

```diff
@@ -279,6 +290,21 @@ private void CopyDefaultValueConstant(ParameterInfo from, ParameterBuilder to)
 					}
 				}
 
+				// Finally, we might have got here because the metadata constant simply doesn't match
+				// the parameter type exactly. Some code generators other than the .NET compilers
+				// might produce such metadata. Make a final attempt to coerce it to the required type:
+				try
+				{
+					var coercedDefaultValue = Convert.ChangeType(defaultValue, parameterNonNullableType);
```

Comment:
Should probably explicitly specify the invariant culture here, otherwise this performs a conversion using the thread's current culture.